src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java

```diff
@@ -153,10 +153,10 @@ protected String name() {
          * in the DFS stack already, there must be a circle in the graph. To speed up the
          * search, all the parsed nodes will be removed from the graph.
          */
-        public boolean hasCycle(ArrayMap<View, ArrayList<View>> dependencyGraph) {
-            ArrayList<View> dfsStack = new ArrayList<View>();
+        public boolean hasCycle(TreeMap<View, List<View>> dependencyGraph) {
```

Comment:
ArrayMap isn't available in API 16, and I needed another map with a concept of "first element"